nmpolicy/internal/capture/result.go

```diff
@@ -17,17 +17,45 @@
 package capture
 
 import (
+	"fmt"
+
 	"github.com/nmstate/nmpolicy/nmpolicy/internal/resolver"
 	"github.com/nmstate/nmpolicy/nmpolicy/types"
 )
 
 type Result struct {
+	lexer          Lexer
+	parser         Parser
+	resolver       Resolver
 	resolverResult resolver.Result
 }
 
-func (r Result) ResolveCaptureEntryPath(
-	capturePath string) (interface{}, error) {
-	return nil, nil
+func NewResult(l Lexer, p Parser, r Resolver, resolverResult resolver.Result) Result {
+	return Result{
+		lexer:          l,
+		parser:         p,
+		resolver:       r,
+		resolverResult: resolverResult,
+	}
+}
+
+func (r Result) ResolveCaptureEntryPath(captureEntryPathExpression string) (interface{}, error) {
+	captureEntryPathTokens, err := r.lexer.Lex(captureEntryPathExpression)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve capture entry path expression: %v", err)
+	}
+
+	captureEntryPathAST, err := r.parser.Parse(captureEntryPathTokens)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve capture entry path expression: %v", err)
+	}
+
+	resolvedCaptureEntryPath, err := r.resolver.ResolveCaptureEntryPath(captureEntryPathAST, r.resolverResult.Unmarshaled)
```

Comment:
The old cached captures don't have `Unmarshaled` version. I would keep the code as it was before the PR. `Capture.Resolve` returns just the marshaled version. Whoever needs to unmarshal it can do it. The added optimization comlicates the code and adds potential bugs (like the one we have now with the cache).

Reply:
So you prefer we re-unmarshal the thing ?

Reply:
I will do that on a follow up